ipn/prefs.go

```diff
@@ -467,6 +474,106 @@ func (p *Prefs) SetAdvertiseExitNode(runExit bool) {
 		netaddr.IPPrefixFrom(netaddr.IPv6Unspecified(), 0))
 }
 
+// peerWithTailscaleIP returns the peer in st with the provided
+// Tailscale IP.
+func peerWithTailscaleIP(st *ipnstate.Status, ip netaddr.IP) (ps *ipnstate.PeerStatus, ok bool) {
+	for _, ps := range st.Peer {
+		for _, ip2 := range ps.TailscaleIPs {
+			if ip == ip2 {
+				return ps, true
+			}
+		}
+	}
+	return nil, false
+}
+
+func checkIPNotSelf(st *ipnstate.Status, ip netaddr.IP) bool {
+	for _, selfIP := range st.TailscaleIPs {
+		if ip == selfIP {
+			return false
+		}
+	}
+	return true
+}
+
+// ClearExitNodeInfo sets the ExitNodeID and ExitNodeIP to their zero values.
+func (p *Prefs) ClearExitNodeInfo() {
+	p.ExitNodeID = ""
+	p.ExitNodeIP = netaddr.IP{}
+}
+
+type ExitNodeLocalIPError struct {
+	hostOrIP string
+}
+
+func (e ExitNodeLocalIPError) Error() string {
+	return fmt.Sprintf("cannot use %s as an exit node as it is a local IP address to this machine", e.hostOrIP)
+}
+
+func validateExitNodeIP(s string, st *ipnstate.Status) (ip netaddr.IP, err error) {
```

Comment:
keep the old func name? The test you moved is still named `TestExitNodeIPOfArg` after all.
 
Not sure "validate" adds much when the signature already returns an error. Validation is implicit.